File: src/app/gkill/api/req_res/get_shared_mi_tasks_response.go
package req_res

import (
	"github.com/mt3hr/gkill/src/app/gkill/api/message"
	"github.com/mt3hr/gkill/src/app/gkill/dao/reps"
)

type GetSharedKyousResponse struct {
	Messages []*message.GkillMessage `json:"messages"`

	Errors []*message.GkillError `json:"errors"`

	MiKyous []*reps.Kyou `json:"mi_kyous"`

	Title string `json:"title"`

	Mis []*reps.Mi `json:"mis"`

	Tags []*reps.Tag `json:"tags"`

	Texts []*reps.Text `json:"texts"`

	TimeIss []*reps.TimeIs `json:"timeiss"`
}

File: src/classes/api/req_res/get-shared-mi-tasks-response.ts
'use strict'

import { Kyou } from '@/classes/datas/kyou'
import { GkillAPIResponse } from '../gkill-api-response'
import type { Text } from '@/classes/datas/text'
import type { Mi } from '@/classes/datas/mi'
import type { Tag } from '@/classes/datas/tag'
import { TimeIs } from '@/classes/datas/time-is'

export class GetSharedMiTasksResponse extends GkillAPIResponse {
    title: string
    mi_kyous: Array<Kyou>
    mis: Array<Mi>
    tags: Array<Tag>
    texts: Array<Text>
    timeiss: Array<TimeIs>

    constructor() {
        super()
        this.title = ""
        this.mi_kyous = new Array<Kyou>()
        this.mis = new Array<Mi>()
        this.tags = new Array<Tag>()
        this.texts = new Array<Text>()
        this.timeiss = new Array<TimeIs>()
    }
}


